src/fr/poseidonscans/src/eu/kanade/tachiyomi/extension/fr/poseidonscans/PoseidonScans.kt

```diff
@@ -1,14 +1,569 @@
 
-class PoseidonScans : Madara(
-    "PoseidonScans",
-    "https://poseidonscans.fr",
-    "fr",
-    dateFormat = SimpleDateFormat("dd/mm/yyyy", Locale.FRANCE),
-) {
-    override val useNewChapterEndpoint = true
+class PoseidonScans : ParsedHttpSource() {
+
+    override val name = "Poseidon Scans"
+    override val baseUrl = "https://poseidonscans.fr"
+    override val lang = "fr"
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient
+    private val json: Json by injectLazy()
+
+    private fun String.toAbsoluteUrl(): String {
+        return if (this.startsWith("http")) this else baseUrl + this
+    }
+
+    private fun String.toApiCoverUrl(): String {
+        if (this.startsWith("http")) return this
+        if (this.contains("storage/covers/")) return "$baseUrl/api/covers/${this.substringAfter("storage/covers/")}"
+        if (this.startsWith("/api/covers/")) return baseUrl + this
+        if (this.startsWith("/")) return baseUrl + this
+        return "$baseUrl/api/covers/$this"
+    }
+
+    @kotlinx.serialization.Serializable
+    data class LatestApiManga(
+        val title: String,
+        val slug: String,
+        val coverImage: String?,
+    )
+
+    @kotlinx.serialization.Serializable
+    data class LatestApiResponse(
+        val success: Boolean,
+        val data: List<LatestApiManga> = emptyList(),
+        val total: Int? = null,
+    )
+
+    @kotlinx.serialization.Serializable
+    data class MangaDetailsData(
+        val title: String,
+        val slug: String,
+        val description: String?,
+        val coverImage: String?,
+        val type: String?,
+        val status: String?,
+        val artist: String?,
+        val author: String?,
+        val alternativeNames: String?,
+        val categories: List<CategoryData>? = emptyList(),
+        val chapters: List<ChapterData>? = emptyList(),
+    )
+
+    @kotlinx.serialization.Serializable
+    data class CategoryData(val name: String)
+
+    @kotlinx.serialization.Serializable
+    data class ChapterData(
+        val number: Float,
+        val title: String? = null,
+        val createdAt: String,
+        val isPremium: Boolean? = false,
+    )
+
+    @kotlinx.serialization.Serializable
+    data class PageImageUrlData(
+        val originalUrl: String,
+        val order: Int,
+    )
+
+    private val isoDateFormatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.ENGLISH).apply {
+        timeZone = TimeZone.getTimeZone("UTC")
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/api/manga/lastchapters?limit=16&page=$page", headers)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val responseBody = response.body.string()
+        val apiResponse = try {
+            json.decodeFromString<LatestApiResponse>(responseBody)
+        } catch (e: Exception) {
+            return MangasPage(emptyList(), false)
+        }
```

Comment:
```suggestion
        val apiResponse = response.parseAs<LatestApiResponse>()
```
`keiyoushi.utils.parseAs`